File: homework01/caesar.py
def encrypt_caesar(plaintext: str) -> str:
    """
    >>> encrypt_caesar("PYTHON")
    'SBWKRQ'
    >>> encrypt_caesar("python")
    'sbwkrq'
    >>> encrypt_caesar("Python3.6")
    'Sbwkrq3.6'
    >>> encrypt_caesar("")
    ''
    """
    ciphertext = ''
    for ch in plaintext:

        if not ch.isalpha():
            ciphertext += ch
        elif 'x' <= ch <= 'z' or 'X' <= ch <= 'Z':
            ciphertext += chr(ord(ch) - 23)
        else:
            ciphertext += chr(ord(ch) + 3)

    return ciphertext


def decrypt_caesar(ciphertext: str) -> str:
    """
    >>> decrypt_caesar("SBWKRQ")
    'PYTHON'
    >>> decrypt_caesar("sbwkrq")
    'python'
    >>> decrypt_caesar("Sbwkrq3.6")
    'Python3.6'
    >>> decrypt_caesar("")
    ''
    """
    plaintext = ''
    for ch in ciphertext:

        if not ch.isalpha():
            plaintext += ch
        elif 'A' <= ch <= 'C' or 'a' <= ch <= 'c':
            plaintext += chr(ord(ch) + 23)
        else:
            plaintext += chr(ord(ch) - 3)

    return plaintext

File: homework01-go/caesar/caesar.go
package caesar

func EncryptCaesar(plaintext string, shift int) string {
	var ciphertext string

	for _, ch := range plaintext {
		if (ch >= 'A' && ch <= 'Z') || (ch >= 'a' && ch <= 'z') {
			if (ch+rune(shift) > 'Z' && ch <= 'Z') || (ch+rune(shift) > 'z') {
				ch = ch - 26
			}
			ciphertext += string(ch + rune(shift))
		} else {
			ciphertext += string(ch)
		}
	}

	return ciphertext
}

func DecryptCaesar(ciphertext string, shift int) string {
	var plaintext string

	for _, ch := range ciphertext {
		if (ch >= 'A' && ch <= 'Z') || (ch >= 'a' && ch <= 'z') {
			if (ch-rune(shift) < 'a' && ch >= 'a') || (ch-rune(shift) < 'A') {
				ch = ch + 26
			}
			plaintext += string(ch - rune(shift))
		} else {
			plaintext += string(ch)
		}
	}

	return plaintext
}
